### ISSUE TITLE
proposal: `prefer_merging_cases`

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/linter#5078]
## Description

Prefer merging switch cases with identical bodies to improve code readability and reduce redundancy.

## Details

Merging switch cases with identical bodies improves code readability and reduces redundancy.

## Kind

This lint enforces style advice by promoting code readability and reducing redundancy through the merging of identical switch case bodies.

## Bad Examples

```dart
switch (value) {
  case value1:
    return 1;   // Lint here
  case value2:
    return 3;
  case value3:
    return 1;   // Lint here
  default:
    return 2;
}
```

## Good Examples 

```dart
switch (value) {
  case value1:
  case value3:
    return 1;
  case value2:
    return 3;
  default:
    return 2;
}
```

```dart
switch (value) {
  case value1 || value3:
    return 1;
  case value2:
    return 3;
  default:
    return 2;
}
```

## Discussion

The future quick-fixes could be the two above examples where the `case value3` is moved up removing the current duplicated code or for it to merge with the other case using the `||` operator.

I believe if this is implemented it should be added to `Effective Dart`.

### Discussion checklist

- [ ] List any existing rules this proposal modifies, complements, overlaps or conflicts with.
- [ ] List any relevant issues (reported here, the [SDK Tracker], or elsewhere).
- [ ] If there's any prior art (e.g., in other linters), please add references here.
- [x] If this proposal corresponds to [Effective Dart] or [Flutter Style Guide] advice, please call it out. (If there isn't any corresponding advice, should there be?)
- [ ] If this proposal is motivated by real-world examples, please provide as many details as you can.  Demonstrating potential impact is _especially valuable_.

**Comment by FMorschel** [comment]
Inspired by https://github.com/dart-lang/sdk/issues/56597

**Comment by FMorschel** [comment]
This would be pretty useful in contexts where there is a big switch statement and some refactoring was done. So that the user can see the identical body.

I know this could lead to false negatives but I think in this case it is better to have some form or warning than to not have any.

**Comment by FMorschel** [comment]
This would be great with https://github.com/dart-lang/sdk/issues/56180 and would closely relate to https://github.com/dart-lang/linter/issues/5079

**Comment by FMorschel** [comment]
Also, https://github.com/dart-lang/linter/issues/5079 would help ensure duplicates are removed as well as `unreachable_switch_case`.
